pkg/cluster/spec/server_config.go

```diff
@@ -218,6 +219,19 @@ func merge2Toml(comp string, global, overwrite map[string]interface{}) ([]byte,
 	return buf.Bytes(), nil
 }
 
+func encodeRemoteCfg2Yaml(remote Remote) ([]byte, error) {
+	buf := bytes.NewBufferString("")
+	enc := yaml.NewEncoder(buf)
+	err := enc.Encode(remote)
+	if err != nil {
+		return nil, err
+	}
+	if buf.String() == "{}\n" {
```

Comment:
This heavily depends on how yaml lib deal with empty struct.

What about `if len(remote.RemoteWrite) == 0 && len(RemoteRead) == 0`?

Reply:
fixed and tested